src/Sulu/Bundle/ContentBundle/Command/MaintainResourceLocatorCommand.php

```diff
@@ -65,93 +110,84 @@ private function upgradeLocale(Webspace $webspace, Localization $localization, O
     {
         $output->writeln('  > Upgrade Locale: ' . $localization->getLocalization('-'));
 
-        /** @var ContentMapperInterface $contentMapper */
-        $contentMapper = $this->getContainer()->get('sulu.content.mapper');
-        $startPage = $contentMapper->loadStartPage($webspace->getKey(), $localization->getLocalization());
+        $contentNode = $this->liveSession->getNode($this->sessionManager->getContentPath($webspace->getKey()));
 
-        $this->upgradeNode($startPage, $webspace, $localization, $output);
-        $this->upgradeByParent($startPage, $webspace, $localization, $contentMapper, $output);
+        $this->upgradeNode($contentNode, $webspace, $localization, $output);
+        $this->upgradeByParent($contentNode, $webspace, $localization, $output);
     }
 
     private function upgradeByParent(
-        StructureBridge $parent,
+        NodeInterface $parentNode,
         Webspace $webspace,
         Localization $localization,
-        ContentMapperInterface $contentMapper,
         OutputInterface $output
     ) {
-        $pages = $contentMapper->loadByParent(
-            $parent->getUuid(),
-            $webspace->getKey(),
-            $localization->getLocalization(),
-            1
-        );
-
-        foreach ($pages as $page) {
-            $this->upgradeNode($page, $webspace, $localization, $output, substr_count($page->getPath(), '/'));
-            $this->upgradeByParent($page, $webspace, $localization, $contentMapper, $output);
+        foreach ($parentNode->getNodes() as $childNode) {
+            $this->upgradeNode($childNode, $webspace, $localization, $output, substr_count($childNode->getPath(), '/'));
+            $this->upgradeByParent($childNode, $webspace, $localization, $output);
         }
     }
 
     private function upgradeNode(
-        StructureBridge $page,
+        NodeInterface $node,
         Webspace $webspace,
         Localization $localization,
         OutputInterface $output,
         $depth = 0
     ) {
-        /** @var SessionManagerInterface $sessionManager */
-        $sessionManager = $this->getContainer()->get('sulu.phpcr.session');
-        $session = $sessionManager->getSession();
-        $node = $session->getNodeByIdentifier($page->getUuid());
+        $locale = $localization->getLocale();
 
-        /** @var RlpStrategyInterface $strategy */
-        $strategy = $this->getContainer()->get('sulu.content.rlp.strategy.tree');
+        $localizedTemplatePropertyName = $this->propertyEncoder->localizedSystemName('template', $locale);
+        if (!$node->hasProperty($localizedTemplatePropertyName)) {
```

Comment:
@chirimoya I have to check for the template, because if there is no template I don't know the URL field at all.